### PR TITLE
5386: Display other petitioners with petitioners

### PR DESCRIPTION
**Pull request by TomElliottFlexion**
With two otherPetitioners: 
Screen Shot 2021-04-16 at 5 21 35 PM

With LOTS of otherPetitioners, collapsed: 
Screen Shot 2021-04-16 at 5 22 11 PM

With LOTS of otherPetitioners, expanded: 
Screen Shot 2021-04-16 at 5 22 35 PM

### CODE REVIEW

File: web-client/src/views/CaseDetail/PetitionerInformation.jsx [hunk 1]
```diff
@@ -182,75 +182,64 @@ const PetitionerInformation = connect(
...
+                  caseInformationHelper.formattedOtherPetitioners.map(
+                    (otherPetitioner, idx) => (
+                      <div
+                        className={classNames(
+                          'grid-col-3',
+                          idx > 1 &&
+                            formattedCaseDetail.contactSecondary &&
+                            'margin-top-8',
+                          idx > 2 &&
+                            !formattedCaseDetail.contactSecondary &&
```
Comment (TomElliottFlexion) [inline web-client/src/views/CaseDetail/PetitionerInformation.jsx]: Redundant?

File: web-client/src/views/CaseDetail/PetitionerInformation.jsx [hunk 2]
```diff
@@ -182,75 +182,64 @@ const PetitionerInformation = connect(
...
+                            </p>
+                            {otherPetitioner.serviceIndicator}
+                          </div>
+                        )}
                       </div>
-                    )}
+                    ),
+                  )}
+              </div>
+              {formattedCaseDetail.petitioners.length > 4 && (
```
Comment (TomElliottFlexion) [inline web-client/src/views/CaseDetail/PetitionerInformation.jsx:165]: Move this check to the caseInformationHelper?

File: web-client/src/presenter/computeds/caseInformationHelper.js [hunk 3]
```diff
@@ -40,6 +37,15 @@ export const caseInformationHelper = (get, applicationContext) => {
...
   const showContactSecondaryEmail = contactSecondary?.email;
 
+  let formattedOtherPetitioners;
+  if (showingAdditionalPetitioners) {
+    formattedOtherPetitioners = otherPetitioners;
+  } else if (contactSecondary) {
+    formattedOtherPetitioners = otherPetitioners.slice(0, 2);
+  } else {
+    formattedOtherPetitioners = otherPetitioners.slice(0, 3);
+  }
```
Comment (rachaelparris) [inline web-client/src/presenter/computeds/caseInformationHelper.js]: I think this block and the JSX could be simplified a bit if we made a single array of everything that should display in that petitioners box, then sliced the first 4 off of it for initial display. Or is there another reason to keep this array separate from the contact primary and secondary for display purposes? I'd be happy to pair or mob on this!

Comment (TomElliottFlexion) [reply web-client/src/presenter/computeds/caseInformationHelper.js]: Excellent idea!

File: web-client/src/presenter/computeds/caseInformationHelper.js [hunk 4]
```diff
@@ -40,20 +32,27 @@ export const caseInformationHelper = (get, applicationContext) => {
   const showContactPrimaryEmail = contactPrimary?.email;
   const showContactSecondaryEmail = contactSecondary?.email;
 
+  const allPetitioners = caseDetail.petitioners;
+
+  const formattedPetitioners = showingAdditionalPetitioners
+    ? allPetitioners
+    : allPetitioners.slice(0, 4);
+
+  console.log('formattedPetitioners', formattedPetitioners);
```
Comment (sutt0n) [inline web-client/src/presenter/computeds/caseInformationHelper.js]: nit: console.log

Comment (TomElliottFlexion) [reply web-client/src/presenter/computeds/caseInformationHelper.js]: Oops, thanks

File: web-client/src/views/CaseDetail/PetitionerInformation.jsx [hunk 5]
```diff
@@ -62,195 +62,126 @@ const PetitionerInformation = connect(
...
-                            aria-label="has e-access"
-                            className="margin-left-05 fa-icon-blue"
-                            icon="flag"
-                            size="1x"
-                          />
+              <div className="grid-row" id="petitioner-information">
+                {caseInformationHelper.formattedPetitioners &&
+                  caseInformationHelper.formattedPetitioners.map(
+                    (petitioner, idx) => (
+                      <div
```
Comment (sutt0n) [inline web-client/src/views/CaseDetail/PetitionerInformation.jsx:69]: nit: i think react will yell at us for not using `key` here

**Review by sutt0n — APPROVED** [review]

**Review by rachaelparris — APPROVED** [review]

**Pull request merged**